### PR TITLE
Add Deferred(DispatchQueue*)

### PR DESCRIPTION
**Pull request by hham**
Add Deferred(DispatchQueue*) and a dedicated unit test.

### CODE REVIEW

File: src/autowiring/test/AutoFilterTest.cpp [hunk 1]
```diff
@@ -664,6 +664,39 @@ TEST_F(AutoFilterTest, DeferredDecorateOnly) {
   ASSERT_EQ(105, dec->i) << "Deferred decorate-only AutoFilter did not properly attach before context termination";
 }
 
+class DQueueSharedPointer:
+public DispatchQueue
```
Comment (codemercenary) [inline src/autowiring/test/AutoFilterTest.cpp]: Tab this one out

**Comment by codemercenary** [comment]
Good call, @hham.  This should guard use cases in Platform.

File: autowiring/Deferred.h [hunk 2]
```diff
@@ -13,4 +14,5 @@ class CoreThread;
 class Deferred {
 public:
   Deferred(CoreThread* pThread) {}
```
Comment (codemercenary) [inline autowiring/Deferred.h]: You can remove this ctor completely.

File: src/autowiring/test/AutoFilterTest.cpp [hunk 3]
```diff
@@ -664,6 +664,39 @@ TEST_F(AutoFilterTest, DeferredDecorateOnly) {
...
+{
+public:
+  DQueueSharedPointer(void)
+  {}
```
Comment (codemercenary) [inline src/autowiring/test/AutoFilterTest.cpp]: You can remove this ctor, too.

File: src/autowiring/test/AutoFilterTest.cpp [hunk 4]
```diff
@@ -664,6 +664,39 @@ TEST_F(AutoFilterTest, DeferredDecorateOnly) {
...
+    callCount++;
+    return Deferred(this);
+  }
+};
+
+TEST_F(AutoFilterTest, DQueueAutoFilterTest) {
+  AutoRequired<AutoPacketFactory> factory;
+  auto ptr = std::make_shared<int>(1012);
+
+  auto dQueueSharedPtr = AutoRequired<DQueueSharedPointer>();
```
Comment (codemercenary) [inline src/autowiring/test/AutoFilterTest.cpp]: Why not `AutoRequired<DQueueSharedPointer> dQueueSharedPtr`?

File: src/autowiring/test/AutoFilterTest.cpp (hunk 1 above)
Comment (hham) [inline src/autowiring/test/AutoFilterTest.cpp]: Done

File: src/autowiring/test/AutoFilterTest.cpp (hunk 3 above)
Comment (hham) [inline src/autowiring/test/AutoFilterTest.cpp]: Done

File: src/autowiring/test/AutoFilterTest.cpp (hunk 4 above)
Comment (hham) [inline src/autowiring/test/AutoFilterTest.cpp]: Done

File: autowiring/Deferred.h (hunk 2 above)
Comment (hham) [inline autowiring/Deferred.h]: Done

File: src/autowiring/test/AutoFilterTest.cpp [hunk 5]
```diff
@@ -664,6 +664,36 @@ TEST_F(AutoFilterTest, DeferredDecorateOnly) {
...
+  auto ptr = std::make_shared<int>(1012);
+
+  AutoRequired<DQueueSharedPointer> dQueueSharedPtr;
+
+  // Generate a bunch of packets, all with the same shared pointer decoration:
+  for(size_t i = 0; i < 100; i++) {
+    // Decorate the packet, forget about it:
+    auto packet = factory->NewPacket();
+    packet->Decorate(ptr);
+  }
```
Comment (codemercenary) [inline src/autowiring/test/AutoFilterTest.cpp]: You probably want to add `ASSERT_EQ(0UL, dQueueSharedPtr->callCount)` at this point, so we can be assured that the filter is not called prematurely.

Comment (codemercenary) [inline src/autowiring/test/AutoFilterTest.cpp]: This is the last thing, then I'll merge.

**Pull request merged**